### PR TITLE
Convert question servers to ESM

### PR DESCRIPTION
**Pull request by jonatanschroeder**

### CODE REVIEW

File: apps/prairielearn/src/question-servers/calculation-worker.js
Comment (nwalters512) [inline apps/prairielearn/src/question-servers/calculation-worker.js:1]: I'm a bit hesitant to convert this file; v2 questions rely pretty specifically on non-ESM stuff for loading v2 question code (`requirejs`). I'd be inclined to update this file to use a `.cjs` extension and document that this shouldn't be CJS.

If you can prove that v2 questions work fine with native ESM, I'm happy to make this change.

Comment (jonatanschroeder) [reply apps/prairielearn/src/question-servers/calculation-worker.js:1]: Does this make a difference?

https://github.com/PrairieLearn/PrairieLearn/blob/1cb19129065b3c38fd1d986e8fb0da16df3c68e4/apps/prairielearn/src/question-servers/calculation-worker.js#L11-L14

I'm fine with keeping this function out of this PR, though.

Comment (jonatanschroeder) [reply apps/prairielearn/src/question-servers/calculation-worker.js:1]: I tested the v2 questions in the test course and they seem to be working fine with ESM code. I also made a change in the code (I hardcoded the correct answer to a question), and the change was ignored before building and picked up properly after building, which confirms the fact that these questions are using the transpiled version. For completeness I also enabled the 'process-questions-in-worker' feature and saw the same behaviour.

Comment (nwalters512) [reply apps/prairielearn/src/question-servers/calculation-worker.js:1]: Did you specifically test with _native_ ESM though? Remember that currently all this is being transpiled back to CJS, but very soon we'll be using native ESM. My concern is specifically about the point in time when we flip that switch. To test that, you can use a `.mjs` extension for this file.

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**